src/app/icd/client/CheckInDelegate.h

```diff
@@ -39,6 +42,62 @@ class DLL_EXPORT CheckInDelegate
                                node that sent the check-in message.
      */
     virtual void OnCheckInComplete(const ICDClientInfo & clientInfo) = 0;
+
+    /**
+     * @brief Callback used to let the application know that a key refresh is
+     * needed to avoid counter rollover problems.
+     *
+     * The implementer of this function should dynamically create a new ICDRefreshKeyInfo object. They should also generate a new
```

Comment:
"dynamically" is an implementation detail.  If they want to use a pool, or a static ICDRefreshKeyInfo, they should be able to.

The key is that the ICDRefreshKeyInfo needs to be tied to a specific key refresh process, right?

Reply:
Yeah. I will modify the description without any implementation details.